addons/hr/models/hr_employee.py

```diff
@@ -535,6 +535,44 @@ def _get_tz(self):
                self.company_id.resource_calendar_id.tz or\
                'UTC'
 
+    def _get_calendar_tz(self, date_from=None):
+        # same as ``_get_tz`` but prioritze the date on the employee's calendar
+        self.ensure_one()
+        if date_from:  # contract overload has different signature
+            calendar = self._get_calendar(date_from)
+        else:
+            calendar = self._get_calendar()
+        return calendar.tz or\
+               self.tz
+
+    @api.model
+    def _get_timezones_query(self):
+        """ Return an sql query (as string) for a `employee_id | timezone` table.
+            Prioritize timezone on employee calendar then on resource
+            This query should match the logic implemented by ``_get_calendar_tz``
+        """
+        return '''(
+            SELECT employee.id AS employee_id,
+                   coalesce(
+                        employee_calendar.tz,
+                        company_calendar.tz,
+                        /* fallback should only be reached in pathological cases
+                         * because the company should have a calendar */
+                        res.tz
+                    ) AS timezone
+              FROM hr_employee employee
+         LEFT JOIN res_company company
+                ON company.id = employee.company_id
+         /* 2 to separate joins instead of 1 with a complex OR clause
+          * to help performance */
```

Comment:
don't write your comment in your sql string :( It's not a good idea for the readability